flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/table/catalog/impl/PulsarCatalogSupport.java

```diff
@@ -198,15 +199,24 @@ public CatalogTable getTable(ObjectPath tablePath) throws PulsarAdminException {
                         table.getComment(),
                         table.getPartitionKeys(),
                         fillDefaultOptionsFromCatalogOptions(table.getOptions()));
+            } catch (PulsarAdminException.NotFoundException e) {
+                throw e;
             } catch (Exception e) {
-                e.printStackTrace();
                 throw new CatalogException(
-                        "Failed to fetch metadata for explict table: " + tablePath.getObjectName());
+                        "Failed to fetch metadata for explict table: " + tablePath.getObjectName(),
+                        e);
             }
         } else {
             String existingTopic = findTopicForNativeTable(tablePath);
-            final SchemaInfo pulsarSchema = pulsarAdminTool.getPulsarSchema(existingTopic);
-            return schemaToCatalogTable(pulsarSchema, existingTopic);
+            try {
+                final SchemaInfo pulsarSchema = pulsarAdminTool.getPulsarSchema(existingTopic);
+                return schemaToCatalogTable(pulsarSchema, existingTopic);
```

Comment:
what about writing a log line with the Schema that we detected.
it is very useful for debugging issues in production